### PR TITLE
Update sway-lib-core to meet SRC-2 inline docs standard

### PR DESCRIPTION
**Pull request by bitzoic**
## Description

This completes the never.sw, ops.sw, primitives.sw, and primitive_conversions.sw tasks of https://github.com/FuelLabs/sway/issues/4812

## Checklist

- [x] I have linked to any relevant issues.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [x] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [x] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: sway-lib-core/src/primitive_conversions.sw [hunk 1]
```diff
@@ -141,6 +235,50 @@ impl u64 {
...
+    /// ```sway
+    /// use std::assert::assert;
+    ///
+    /// fn foo() {
+    ///     let val = 10u32;
+    ///     let result = val.as_u64();
+    ///     assert(result == 10);
+    /// }
+    /// ```
+    pub fn as_u64(self) -> u64 {
```
Comment (SwayStar123) [inline sway-lib-core/src/primitive_conversions.sw:255]: is this a new function added? Should be in a seperate pr with testing

Comment (bitzoic) [reply sway-lib-core/src/primitive_conversions.sw:255]: Oh no... I think this is an artifact from the master branch

**Comment by SwayStar123** [comment]
Test still fails

**Comment by bitzoic** [comment]
> Test still fails

Just need to merge in the `standard-documentation` branch

**Review by SwayStar123 — APPROVED** [review]

**Pull request merged**